### ISSUE TITLE
Invalid `const cannot be reassigned` errors

### ISSUE THREAD

**Issue by Gozala** [facebook/flow#844]
I seem to get invalid `const cannot be reassigned` error for `const` uses with in the function body:

```js
/* @flow */

type List = {head:any, tail:List};
const second = (list:List) => {
  const [head, tail] = list;
  return tail;
}
```

**Comment by sporto** [comment]
I'm getting the same error, with:

```
const { value } = event.target
```

flow 0.16

**Comment by samwgoldman** [comment]
Should be fixed by #831

**Comment by samwgoldman** [comment]
Fixed in e7ea4dc74bb79b62c2b707ceeeacf7ddf102147c

**Comment by Gozala** [comment]
Is this going to be released anytime soon ? I'd like to release version of my library that uses flow via travis.ci but current stable version 0.16.0 can't handle many cost usages.
